routers/web/web.go

```diff
@@ -108,7 +108,15 @@ func Routes(ctx gocontext.Context) *web.Route {
 	routes := web.NewRoute()
 
 	routes.Head("/", misc.DummyOK) // for health check - doesn't need to be passed through gzip handler
-	routes.RouteMethods("/assets/*", "GET, HEAD", CorsHandler(), public.AssetsHandlerFunc("/assets/"))
+
+	if setting.IsProd {
+		routes.RouteMethods("/assets/*", "GET, HEAD", CorsHandler(), public.AssetsHandlerFunc("/assets/"))
+	} else {
+		routes.RouteMethods("/assets/*", "GET, HEAD", CorsHandler(),
+			public.MakeDevAssetsHandler("http://localhost:3874", "/assets"),
```

Comment:
It would cause problems for developers if they already have something working on port 3874

Reply:
The same is true for default port 3000 which gitea listens on. Could add an environment variable to share and change this port.

Reply:
I think we can make dev server use dynamic port and write the port into a file, then gitea can get the dynamic port and use it.

Reply:
I think that's overkill and it would introduce a certain dependency between the two I don't like. Static port is fine as long as its changeable.

Reply:
As long as the "web dev server" and "backend proxy" are introduced, the "dependency" and "overkill" are there and they can't be avoided.

The question is: how to make it friendly to all developers and make the overkill as minimal as possible.

The approaches:

1. Use fixed web dev server port
    * It would cause problems for developers who have run programs on that port
2. Use "environment variable" to set a web dev server port
    * It's not friendly to developers who do not use "make watch", for example, I always use Goland to start the debugger, no command line is involved.
    * It's not friendly to Windows developers, the Windows environment variables are always tricky.
3. Use "app.ini" to set a web dev server port
    * JS code could do a simple grep to get the port and use it for web dev server
4. Make web dev server use a dynamic port and tell Gitea to use this port
    * A file like `./data/tmp/web-dev-server-port.json` can be used
    * This approach works for all cases, and it's totally transparent to developers because the dynamic port would never conflict.

I think approach 3 is acceptable (some developers till need to learn and manually change something if the default port conflicts) and approach 4 is the best (totally transparent).

Reply:
How about "Use "environment variable" to set a web dev server port" and default the  variable in the makefile. Essentially it forces everyone to use the Makefile but I think that is pretty much a given, or do you not use Makefile at alle for dev tasks?

Reply:
> or do you not use Makefile at all for dev tasks?

I seldom use `make` when developing ....

Reply:
I guess we could use env var like `GITEA_DEV_FRONTEND_PORT` with the fallback to static port in both webpack config and go. That should satisfy all requirements. Seems much easier than attempting a ini parser in JS imho.

Reply:
`GITEA_DEV_FRONTEND_PORT`: I have no objection for doing this, but it might be tricky for Windows developers (not sure because I am not using Windows heavily at the moment).

Reply:
I think any Windows dev ought to develop inside WSL nowadays where they have a full Linux environment available. No one in their right mind would develop natively on Windows (e.g. `cmd`/`powershell`). I know @jolheiser has been doing it in the past, but with git-bash.

Reply:
Then it seems good to try GITEA_DEV_FRONTEND_PORT first. It should be not difficult to switch or improve the approaches.